### PR TITLE
replace Anchore engine with grype

### PR DESCRIPTION
**Pull request by markysoft**
# Description

replace Anchore engine with grype
Add PR template with checklist

# Checklist:

- [x] I have ensured the Defra version in the **JOB.env** file matches that in the **Dockerfile**
- [x] I have ensured the runtime versions in the **image-matrix.json** match the **Dockerfile** and the table in the **README.md**
- [x] I have added newly ignored vulnerabilities to the **POLICY_CONFIGURATION.md**
- [x] I have checked if previously identified vulnerabilities have been patched, and can be removed from the **.grype.yaml** and **POLICY_CONFIGURATION.md** files

### CODE REVIEW

File: .github/workflows/nightly-scan.yml [hunk 1]
```diff
@@ -29,14 +29,18 @@ jobs:
...
+          fail-build: true
+          severity-cutoff: "medium"
 
       - name: Archive anchore report
         if: ${{ failure() }}
         uses: actions/upload-artifact@v2
         with:
-          name: anchore-reports-dotnet-${{ matrix.image.runtimeVersion }}
-          path: anchore-reports
+          name:  anchore-reports-dotnet-${{ matrix.image.runtimeVersion }}
```
Comment (paulsimonandrews) [inline .github/workflows/nightly-scan.yml:45]: did you want to call this one "netcore" instead of "dotnet" too?

**Review by paulsimonandrews — COMMENTED** [review]
Couple of questions:
* Should the empty `.grype` file be called `.grype.yaml`?
* Check the comment about name change of report for nightly scan, dotnet -> netcore

File: .github/workflows/nightly-scan.yml (hunk 1 above)
Comment (markysoft) [reply .github/workflows/nightly-scan.yml:45]: Or vice versa! Thanks, good spot on both, I've updated them now.

**Review by paulsimonandrews — APPROVED** [review]
LGTM

**Pull request merged**